src/site/content/en/learn/pwa/os-integration/index.md

```diff
@@ -1,11 +1,258 @@
 ---
-title: OS integration
-description: >
-  Our app now works outside of the browser. In this chapter, you will understand how to integrate further with the operating system.
+title: OS Integration
 authors:
   - firt
-date: 2021-11-03
-placeholder: true
+description: >
+  Your PWA now works outside the browser. This chapter covers how to integrate further with the operating system once users install your app.
+date: 2022-04-15
 ---
 
-Coming soon!
+Web apps have a big reach. They run on multiple platforms. They are easy to share via links. But traditionally they lacked integration with the operating system. Not long ago they were not even installable. Luckily that has changed and now we can take advantage of that integration to add useful features to our PWAs. Let's explore some of those options.
+
+## Working with the file system
+
+A typical user workflow using files looks like this:
+* Pick a file or folder from the device and open it directly.
+* Make changes to those files or folders, and save the changes back directly.
+* Make new files and folders.
+
+Before the [File System Access API](/file-system-access/), web apps couldn't do this. Opening files required a file upload, saving changes required users to download them, and the web had no access at all to make new files and folders in the user's filesystem.
+
+### Opening a file
+
+To open a file we use the `window.showOpenFilePicker()` method. Note that this method requires a user gesture, such as a button click. Here is the rest of the setup for opening a file:
+* First, capture the [file handle](https://developer.mozilla.org/docs/Web/API/FileSystemHandle) from the file system access's file picker API. This gives you basic information about the file.
```

Comment:
this could be an ordered list like the one below.

Reply:
Done.